packages/grafana-ui/src/components/Combobox/Combobox.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Combobox, Option } from './Combobox';
+
+// Mock data for the Combobox options
+const options: Option[] = [
+    { label: 'Option 1', value: '1' },
+    { label: 'Option 2', value: '2' },
+    { label: 'Option 3', value: '3', description: 'This is option 3' },
+    { label: 'Option 4', value: '4' },
+];
+
+describe('Combobox', () => {
+    const onChangeHandler = jest.fn();
+    beforeAll(() => {
+        const mockGetBoundingClientRect = jest.fn(() => ({
+            width: 120,
+            height: 120,
+            top: 0,
+            left: 0,
+            bottom: 0,
+            right: 0,
+        }));
+
+        Object.defineProperty(Element.prototype, 'getBoundingClientRect', {
+            value: mockGetBoundingClientRect,
+        });
+    });
+
+    it('renders without error', () => {
+        render(<Combobox options={options} value={null} onChange={onChangeHandler} />);
+        expect(screen.getByRole('combobox')).toBeInTheDocument();
+    });
+
+
+    // it('should allow selecting a value by clicking directly', async () => {
+    //     render(<Combobox options={options} onChange={onChangeHandler} value={null} />);
+
+    //     const input = screen.getByRole('combobox');
+    //     userEvent.click(input);
+
+    //     const item = await screen.findByRole('option', { name: 'Option 1' });
+    //     userEvent.click(item);
+    //     expect(screen.queryByDisplayValue('Option 1')).toBeInTheDocument();
```

Comment:
```suggestion
    //     expect(await screen.findByDisplayValue('Option 1')).toBeInTheDocument();
```
The input value doesn't get updated "immediately", so we need to wait for it to appear. It is a bit unclear why this isn't the case for the other test cases. If you do that, this test will pass, so you can uncomment it :)